packages/server/src/questionType/questionType.controller.ts

```diff
@@ -13,54 +13,61 @@ import {
 } from '@nestjs/common';
 import { Roles } from 'decorators/roles.decorator';
 import { JwtAuthGuard } from 'guards/jwt-auth.guard';
-import { Connection } from 'typeorm';
 import { QuestionTypeModel } from './question-type.entity';
 import { Response } from 'express';
+import { QueueRolesGuard } from 'guards/queue-role.guard';
+import { CourseRolesGuard } from 'guards/course-roles.guard';
+import { RolesGuard } from 'guards/role.guard';
 
 @Controller('questionType')
 @UseGuards(JwtAuthGuard)
 @UseInterceptors(ClassSerializerInterceptor)
 export class QuestionTypeController {
-  constructor(private connection: Connection) {}
-
-  @Post(':c')
+  @Post(':courseId')
+  @UseGuards(CourseRolesGuard)
   @Roles(Role.TA, Role.PROFESSOR)
   async addQuestionType(
     @Res() res: Response,
-    @Param('c') courseId: number,
+    @Param('courseId') courseId: number,
     @Body() newQuestionType: QuestionTypeParams,
   ): Promise<void> {
-    console.log('newQuestionType', newQuestionType);
     let queueId = newQuestionType.queueId;
     if (typeof queueId !== 'number' || isNaN(queueId)) {
       queueId = null;
     }
     const questionType = await QuestionTypeModel.findOne({
       where: {
         cid: courseId,
-        queueId: newQuestionType.queueId,
+        queueId: queueId,
         name: newQuestionType.name,
       },
     });
     if (!questionType) {
-      await QuestionTypeModel.create({
-        cid: courseId,
-        name: newQuestionType.name,
-        color: newQuestionType.color,
-        queueId: newQuestionType.queueId,
-      }).save();
-      res.status(200).send('success');
-      return;
+      try {
+        await QuestionTypeModel.create({
+          cid: courseId,
+          name: newQuestionType.name,
+          color: newQuestionType.color,
+          queueId: queueId,
+        }).save();
+        res.status(200).send('success');
+        return;
+      } catch (e) {
+        res.status(400).send('Error creating question type');
+        return;
+      }
     } else {
-      res.status(400).send('Question already exists');
+      res.status(400).send('Question type already exists');
       return;
     }
   }
 
-  @Get(':c/:queueId')
+  @Get(':courseId/:queueId')
+  @UseGuards(QueueRolesGuard)
```

Comment:
Should it be a queue role guard or course role guard?

Reply:
Actually you're right! good catch. When I was adding the guards, a part of me didn't realize that that one can have queueId be not a number. Good thing too, I had to make a new test for this and make some changes.